pkg/autodetect/main_test.go

```diff
@@ -243,6 +238,119 @@ func TestAutoDetectEsProvisionWithEsOperator(t *testing.T) {
 	assert.True(t, viper.GetBool("es-provision"))
 }
 
+func TestAutoDetectKafkaProvisionNoKafkaOperator(t *testing.T) {
+	// prepare
+	defer viper.Reset()
+
+	dcl := &fakeDiscoveryClient{}
+	cl := fake.NewFakeClient()
+	b := WithClients(cl, dcl)
+
+	viper.Set("kafka-provision", v1.FlagProvisionKafkaAuto)
```

Comment:
Should this be above the `defer viper.Reset()` as per the changes made in the es provisioning tests?